datetime/src/time.rs

```diff
@@ -0,0 +1,128 @@
+    let args: Vec<String> = env::args().skip(1).collect();
+
+    if args.is_empty() {
+        return Err(format!("No arguments provided").into());
+    }
+
+    // Initialize the default values
+    let mut posix = false;
+    let mut utility = String::new();
+    let mut arguments = Vec::new();
+    let mut utility_found = false;
+
+    // Parse arguments
+    for arg in args {
+        if !utility_found {
+            if arg == "-p" {
+                posix = true;
+            } else {
+                utility = arg;
+                utility_found = true;
+            }
+        } else {
+            arguments.push(arg);
+        }
+    }
+
+    // Create the Args struct
+    let parsed_args = Args {
+        posix,
+        utility,
+        arguments,
+    };
+
+    Ok(parsed_args)
+}
+
+fn time(args: Args) -> Result<(), Box<dyn std::error::Error>> {
+    let start_time = Instant::now();
+    // SAFETY: std::mem::zeroed() is used to create an instance of libc::tms with all fields set to zero.
+    // This is safe here because libc::tms is a Plain Old Data type, and zero is a valid value for all its fields.
+    let mut tms_start: libc::tms = unsafe { std::mem::zeroed() };
+    // SAFETY: sysconf is a POSIX function that returns the number of clock ticks per second.
+    // It is safe to call because it does not modify any memory and has no side effects.
+    let clock_ticks_per_second = unsafe { libc::sysconf(libc::_SC_CLK_TCK) as f64 };
+
+    // SAFETY: times is a POSIX function that fills the provided tms structure with time-accounting information.
+    // It is safe to call because we have correctly allocated and initialized tms_start, and the function
+    // only writes to this structure.
+    unsafe { libc::times(&mut tms_start) };
+
+    let mut child = Command::new(&args.utility)
+        .args(args.arguments)
+        .stderr(Stdio::piped())
+        .spawn()
+        .map_err(|_| format!("Command not found: {}", args.utility))?;
+
+    let status = child.wait()?;
+
+    let elapsed = start_time.elapsed();
+    let tms_end: libc::tms = unsafe { std::mem::zeroed() };
+
+    let user_time = (tms_start.tms_utime - tms_end.tms_utime) as f64 / clock_ticks_per_second;
+    let system_time = (tms_start.tms_stime - tms_end.tms_stime) as f64 / clock_ticks_per_second;
+
+    if args.posix {
+        writeln!(
+            io::stderr(),
+            "real {:.6}\nuser {:.6}\nsys {:.6}",
+            elapsed.as_secs_f64(),
+            user_time,
+            system_time
+        )?;
+    } else {
+        writeln!(
+            io::stderr(),
+            "Elapsed time: {:.6} seconds\nUser time: {:.6} seconds\nSystem time: {:.6} seconds",
+            elapsed.as_secs_f64(),
+            user_time,
+            system_time
+        )?;
+    }
+
+    std::process::exit(status.code().unwrap_or(1));
+}
+
+fn main() -> Result<(), Box<dyn std::error::Error>> {
+    // parse command line arguments
+    let args = parse()?;
```

Comment:
Use correct exit codes for the utility:
https://pubs.opengroup.org/onlinepubs/009695299/utilities/time.html
Failure at this stage should return 1-125 error code.